atomsci/ddm/pipeline/model_pipeline.py

```diff
@@ -42,40 +56,53 @@ def calc_AD_kmean_dist(train_dset, pred_dset, k, train_dset_pair_distance=None,
         # calculate the pairwise distance of training set
         train_dset_pair_distance = pairwise_distances(X=train_dset, metric=dist_metric)
     train_kmean_dis = []
+    # here we use k+1 since the nearest neighbor is itself.
+    kn_idxes = get_k_nearest_index(train_dset_pair_distance, k+1)
     for i in range(len(train_dset_pair_distance)):
-        kn_idx = np.argpartition(train_dset_pair_distance[i], k+1)
-        dis = np.mean(train_dset_pair_distance[i][kn_idx[:k+1]])
+        # instead of taking the mean, we divide by k. Leaving out the distance to itself
+        dis = np.sum(train_dset_pair_distance[i][kn_idxes[i]])/k
```

Comment:
isn't this equivalent to the mean of non-self distances?

Reply:
Yeah, I'm trying to get the mean of non-self distances. The original code would do something like this for k=3
sum(0, 1, 1.2, 1.3)/4 while the new code does sum(0, 1, 1.2, 1.3)/3.